src/components/ChangePlan/PlanCalculator.js

```diff
@@ -125,15 +143,28 @@ const PlanCalculator = ({
         planList,
         appSessionRef,
       );
-
       if (plansByType.length) {
         setState({
           loading: false,
           planList: plansByType,
           discountsList: plansByType[0].billingCycleDetails?.map(mapDiscount),
           planTypes: planTypes,
           selectedPlanType: selectedPlanType,
-          descriptions: plansByType.map((plan) => plan.name),
+          descriptions: plansByType.map((plan) => {
+            return {
+              amount:
+                plan.type === 'prepaid'
+                  ? plan.credits
+                  : plan.type === 'subscribers'
+                  ? plan.subscriberLimit
+                  : plan.type === 'monthly-deliveries'
+                  ? plan.emailsByMonth
+                  : null,
+              description: _(
+                'plan_calculator.' + plan.type.replace('-', '_') + '_amount_description',
```

Comment:
I'm not so sure about this, think that if someone were to add a new plan and you do not have the resource in the language file, you would see gibberish.

Reply:
I don't understand how can we fix this. In that case I suppose that we have to code the FE before, to anticipate us at the new changes.

Reply:
Or we can get the description with their translation from the BE, but I think that it wouldn't be a good idea.

Reply:
A function would do fine.

Reply:
```
...
descriptionId: planItemDescription(plan.type)
...
const planItemDescription = (planType) => {
   switch (planType): {
      case 'subscribers': 
         return ('plan_calculator.subscribers_amount_description');
      ...
      default: 
         return ('plan_calculator.unknown_amount_description'); // --> this could be empty or default to something else
   }
}
```